spec/ameba/rule/style/heredoc_escape_spec.cr

```diff
@@ -0,0 +1,83 @@
+require "../../../spec_helper"
+
+module Ameba::Rule::Style
+  describe HeredocEscape do
+    subject = HeredocEscape.new
+
+    it "passes if a heredoc doesn't contain interpolation" do
+      expect_no_issues subject, %{
+        <<-HEREDOC
+          foo
+          HEREDOC
+      }
+    end
+
+    it "passes if a heredoc contains interpolation" do
+      expect_no_issues subject, %{
+        <<-HEREDOC
+          foo \#{:bar}
+          HEREDOC
+      }
+    end
+
+    it "passes if a heredoc contains normal and escaped interpolation" do
+      expect_no_issues subject, %{
+        <<-HEREDOC
+          foo \\\#{:bar} \#{:baz}
+          HEREDOC
+      }
+    end
+
+    it "fails if a heredoc contains escaped interpolation" do
+      expect_issue subject, %{
+        <<-HEREDOC
+      # ^^^^^^^^^^ error: Use an escaped heredoc
+          foo \\\#{:bar}
+          HEREDOC
+      }
+    end
+
+    pending "passes if a heredoc contains normal and escaped escape sequences" do
```

Comment:
Need to figure out a way to determine which escape sequences are explicitly specified (i.e., `\n` in text versus an actual newline). The lexer does not give an easy way of doing so.